tket2/src/rewrite_space/factory/kak.rs

```diff
@@ -0,0 +1,255 @@
+                    // the same place. As such we put the searcher
+                    // back on the queue.
+                    } else {
+                        searching_walker.two_qubit_node = Some(expanded_node);
+                        searching_walker_list.push(searching_walker);
+                    }
+                }
+            }
+
+        }
+
+        matches.into_iter()
+    }
+
+    fn op_cost(&self, op: &hugr::ops::OpType) -> Option<Self::Cost> {
+        todo!()
+    }
+    
+    fn get_replacement(
+        &self,
+        pattern_match: &Self::PatternMatch,
+        host: &PersistentHugr,
+    ) -> Option<Hugr> {
+        todo!()
+    }
+    
+    fn map_boundary(
+        &self,
+        node: PatchNode,
+        port: hugr::Port,
+        pattern_match: &Self::PatternMatch,
+        host: &PersistentHugr,
+    ) -> hugr::Port {
+        todo!()
+    }
+    
+    fn get_name(&self, pattern_match: &Self::PatternMatch, host: &PersistentHugr) -> String {
+        todo!()
+    }
+}
+
+/// only call this on nodes that are known quantum gates
+fn n_qubits<N: Copy>(node: N, hugr: &impl HugrView<Node = N>) -> usize {
+    debug_assert_eq!(
+        hugr.signature(node).unwrap().input_count(),
+        hugr.signature(node).unwrap().output_count(),
+        "expected a quantum gate"
+    );
+    hugr.signature(node).unwrap().input_count()
+}
+
+fn empty_2qb_hugr() -> Hugr {
+    let builder = DFGBuilder::new(endo_sig(vec![qb_t(), qb_t()])).unwrap();
+    let [q0, q1] = builder.input_wires_arr();
+    builder.finish_hugr_with_outputs(vec![q0, q1]).unwrap()
+}
+
+fn one_cz_2qb_hugr() -> Hugr {
+    let mut builder = DFGBuilder::new(endo_sig(vec![qb_t(), qb_t()])).unwrap();
+    let [q0, q1] = builder.input_wires_arr();
+    let cz1 = builder.add_dataflow_op(Tk2Op::CZ, vec![q0, q1]).unwrap();
+    let [q0, q1] = cz1.outputs_arr();
+    builder.finish_hugr_with_outputs(vec![q0, q1]).unwrap()
+}
+
+#[test]
+fn test_kak_resynthesis_init() {
+    
+    let pers_hugr = PersistentHugr::with_base(one_cz_2qb_hugr());
+    
+    for node in pers_hugr.nodes() {
+        let walker = Walker::from_pinned_node(node, pers_hugr.as_state_space());
+
+        if pers_hugr.get_optype(node) == &Tk2Op::CZ.into() {
+            println!("{}", node);
+            KAKResynthesis{ 
+                wires: [
+                    vec![walker.get_wire(node, OutgoingPort::from(0))],
+                    vec![walker.get_wire(node, OutgoingPort::from(1))]] 
+            };
+        }
+        
+    }
+}
+
+#[test]
+fn test_find_pattern_matches() {
+    let hugr = one_cz_2qb_hugr();
+
+    println!("{}", hugr.mermaid_string());
+
+    let factory = KAKFactory;
+
+    let pers_hugr = PersistentHugr::with_base(hugr);
+    for node in pers_hugr.nodes() {
+        let walker = Walker::from_pinned_node(node, pers_hugr.as_state_space());
+        // walker.get_wire(node, OutgoingPort::from(0));
+        factory.find_pattern_matches(node, walker);
+    }
```

Comment:
Add an assertion to check this does what you expect.

The simplest would be to add a variable `let mut n_matches = 0` before the loop, and then count the number of matches you find:
```rust
n_matches += factory.find_pattern_matches(node, walker).count();
```

After the loop, you would write `assert_eq(n_matches, 1)`.

A stronger test would involve looking into the matches obtained and checking it contains the gates you expect.